music21/musicxml/m21ToXml.py

```diff
@@ -3472,7 +3521,23 @@ def noteToXml(self, n: note.GeneralNote, noteIndexInChord=0, chordParent=None):
             for t in mxTieList:
                 mxNote.append(t)
 
-        # TODO: instrument
+        # instrument tags are necessary when there is more than one
+        if self.parent is not None and len(self.parent.instrumentStream) > 1 and not n.isRest:
```

Comment:
Does this mean that we'll be doing a getContextByClass for each note or chord being exported if there is more than one instrument in a score?  This might be much slower.

Reply:
We should probably hold this until we handle note storage on `NotRest` in #1033, then I can replace this block with the `getStoredInstrument` lookup which will be cached.